narwhals/_spark_like/dataframe.py

```diff
@@ -173,16 +176,87 @@ def sort(
 
         flat_by = flatten([*flatten([by]), *more_by])
         if isinstance(descending, bool):
-            descending = [descending]
+            descending = [descending] * len(flat_by)
 
         if nulls_last:
-            sort_funcs = [
+            sort_funcs = (
                 F.desc_nulls_last if d else F.asc_nulls_last for d in descending
-            ]
+            )
         else:
-            sort_funcs = [
+            sort_funcs = (
                 F.desc_nulls_first if d else F.asc_nulls_first for d in descending
-            ]
+            )
 
         sort_cols = [sort_f(col) for col, sort_f in zip(flat_by, sort_funcs)]
         return self._from_native_frame(self._native_frame.sort(*sort_cols))
+
+    def with_row_index(self: Self, name: str) -> Self:
+        import pyspark.sql.functions as F  # noqa: N812
+
+        return self._from_native_frame(
+            self._native_frame.withColumn(name, F.monotonically_increasing_id())
+        )
```

Comment:
should we keep this one out for now? i think it would be good to aim to have the lazy layer well-defined, and initially at least to leave out anything that's row-order-dependent

Reply:
`monotonically_increasing_id` will be performant but not deterministic. Do we need `with_row_index` to be deterministic?
I tend to agree with "let's leave this out for now"